perf(linter): remove allocation in `CompositeFix::merge_fixes`

overlookmotel opened the pull request:
`String::shrink_to_fit` will generally reallocate in order to reduce the `String`'s capacity to minimum size. 

In `merge_fixes`, the size of the `String` is unlikely to be excessively over-sized - in best case, it's exactly the right size, and in worst case it's twice as large than it needs to be. The reallocation probably costs more than the gain of clawing back a few bytes of memory. So remove this operation.

camc314 commented:
### Merge activity

* **Sep 24, 7:52 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
perf(linter): remove allocation in `CompositeFix::merge_fixes` (#14090)

`String::shrink_to_fit` will generally reallocate in order to reduce the `String`'s capacity to minimum size.

In `merge_fixes`, the size of the `String` is unlikely to be excessively over-sized - in best case, it's exactly the right size, and in worst case it's twice as large than it needs to be. The reallocation probably costs more than the gain of clawing back a few bytes of memory. So remove this operation.
